core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -1564,6 +1564,50 @@ public void _testLikeAndSimilarFails() {
         .fails("No match found for function signature FOO..");
   }
 
+  @Test void testInvalidTableFunction() {
+    // A table function at most have one input table with row semantics
+    sql("select * from table(^invalid(table orders, table emp)^)")
+        .fails("A table function at most has one input table with row semantics."
+            + " Table function 'INVALID' has multiple input tables with row semantics");
+    // Only tables with set semantics may be partitioned
+    sql("select * from table(^score(table orders partition by productid)^)")
+        .fails("Only tables with set semantics may be partitioned."
+            + " Invalid PARTITION BY clause in the 0-th operand of table function 'SCORE'");
+    // Only tables with set semantics may be ordered
```

Comment:
We need some tests for multiple partition/ordering keys.
We need some tests for multiple table params.

Reply:
> We need some tests for multiple partition/ordering keys.

There are already exist cases to cover these in SqlValidatorTest#testTableFunctionWithTableParam.

> We need some tests for multiple table params.

Based on SQL standard, a table function at most has one input table with row semantics, there are already exist cases to cover these based on MockSqlOperatorTable#InvalidTableFunction.

I add some cases to cover a table function with multiple input tables with set semantics based on MockSqlOperatorTable#SimilarlityTableFunction.